lib/transaction/transaction.js

```diff
@@ -289,6 +300,11 @@ Transaction.prototype.toBufferWriter = function(writer) {
     output.toBufferWriter(writer);
   });
   writer.writeUInt32LE(this.nLockTime);
+
+  if (this.isSpecialTransaction() && this.hasCorrectExtraPayloadSize()) {
+    writer.writeVarintNum(this.extraPayloadSize);
```

Comment:
Discussed that we have to throw an exception here. It's pretty complicated because of old tests, so we decided to add `@private` JSDoc tag to this method.

Reply:
@shumkov Fixed. I assume we need to change bitcore's architecture in the future